spec/grouping-nodes.md

```diff
@@ -0,0 +1,146 @@
+make it possible to define configuration once but apply it to each Node having
+the same Role. Today a DataPlaneNode can only belong to a single Role. Today
+there is CRD field level inheritance defined between Role and Node. If a CRD
+field is defined both in the Role and the Node then the value from the Node
+will take precedence. With today's model the DataPlaneRole has a direct
+grouping effect for the NovaExternalCompute CRs as well.
+
+However if we move the EDPM related OpenStack service creation from the
+dataplane-operator to the respective service-operator [as proposed in a
+separate spec](https://github.com/openstack-k8s-operators/docs/pull/41), then
+DataPlaneRole based grouping will not be automatically applied. So we need a
+different solution.
+
+Moreover the current restriction that a DataPlaneNode can only belong to a
+single DataPlaneRole is limiting. From nova perspective a set of compute nodes
+can be configured with CPU pinning while from neutron perspective a set of
+compute nodes can be configured with DPDK. If the two sets of nodes overlap but
+are not equal then there will be a need for additional roles to be defined for
+each combination of the configuration (pinning with DPDK, pinning without
+DPDK, sharing with DPDK, sharing without DPDK). This shows that the current
+restriction can lead to the explosion of the number of necessary groups to
+model the different combinations of service configs.
+
+## Proposed change
+
+Allow an EDPM node to belong to multiple groups by defining OpenStack service
+specific profiles. The DataPlaneRole could remain as is today but it would
+represent the grouping of the EDPM nodes only just from the generic
+hardware, host OS, infrastructure configuration perspective.
+
+Each service operator that has EDPM specific service CRD (e.g.
+NovaExternalCompute, OVNMetadataAgent) can define a Profile CRD (e.g.
+ComputeProfile, NetworkProfile) to describe the service specific configuration
+of a set of EDPM nodes. Then the human deployer can use annotations on the
+DataPlaneRole or DataPlaneNode to select a single Profile per service type.
+
+### Example
+
+ComputeProfile with shared CPUs
+```yaml
+apiVersion: nova.openstack.org/v1beta1
+kind: ComputeProfile
+metadata:
+    name: shared-cpus
+spec:
+    customComputeServiceConfig: |
+    [compute]
+    cpu_shared_set = 4-12,^8,15
+    [DEFAULT]
+    cpu_allocation_ratio = 4.0
+```
+
+ComputeProfile with pinned CPUs
+```yaml
+apiVersion: nova.openstack.org/v1beta1
+kind: ComputeProfile
+metadata:
+name: dedicated-cpus
+spec:
+    customComputeServiceConfig: |
+    [compute]
+    cpu_dedicated_set = 4-12,^8,15
+    [DEFAULT]
+    cpu_allocation_ratio = 1.0
+```
+
+OpenstackDataPlaneRole/edpm-compute
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneRole
+metadata:
+    name: edpm-compute
+    annotations:
+        nova.openstack.org/cell: nova-cell1
+        nova.openstack.org/compute-profile: shared-cpus
```

Comment:
So since this is a dict and the key is (effectively) compute-profile, we're *not* talking about composing multiple `ComputeProfile`s. I think that's obvious, but maybe worth a note somewhere.